tao_compiler/mlir/xla/ral/BUILD

```diff
@@ -477,22 +477,65 @@ tf_gpu_kernel_library(
 ###########################################################################################
 # The targets following are for tf bridge side build. Get rid of all the deps from tf source code, only use some of the `.bzl` utils from tensorflow.
 
+cc_library(
+    name = "disc_mkl_bridge",
+    hdrs = glob([
+        "context/mkldnn/ideep/*.hpp",
+        "context/mkldnn/ideep/ideep/*.hpp",
+        "context/mkldnn/ideep/ideep/operators/*.hpp",
+    ]),
+    linkopts = [
+        "-L$$(dirname $(locations @onednn//:onednn_lib)",
+        "-Wl,--start-group",
+        "$(location @onednn//:onednn_lib)",
+        "-Wl,--end-group",
+    ] + if_disc_x86([
+        "-L$(location @mkl_static//:lib)/",
+        "-Wl,--start-group",
+        "$(location @mkl_static//:lib)/libmkl_intel_ilp64.a",
+        "$(location @mkl_static//:lib)/libmkl_gnu_thread.a",
+        "$(location @mkl_static//:lib)/libmkl_core.a",
+        "-Wl,--end-group",
+    ]) + if_disc_aarch64([
+        "-L$$(dirname $(location @acl_compute_library//:build/libarm_compute.a)",
+        "-Wl,--start-group",
+        "$(location @acl_compute_library//:build/libarm_compute.a)",
+        "$(location @acl_compute_library//:build/libarm_compute_core.a)",
+        "$(location @acl_compute_library//:build/libarm_compute_graph.a)",
+        "-Wl,--end-group",
+    ]),
+    copts = [
+        "-I$(location @mkl_include//:include)",
+    ],
+    data = [
+        "@mkl_static//:lib",
+        "@mkl_include//:include",
+        "@onednn//:onednn_lib",
+        "@acl_compute_library//:build/libarm_compute.a",
+        "@acl_compute_library//:build/libarm_compute_core.a",
+        "@acl_compute_library//:build/libarm_compute_graph.a",
+    ],
+    deps = [
+        "@onednn//:onednn",
+    ],
+)
+
 cc_library(
     name = "common_context_bridge",
     srcs = [
         "context/common_context_impl.cc",
-    ] + if_cuda_or_rocm([
         "context/common_context_impl_cuda.cc",
         "context/stream_executor_based_impl.cc",
+    ] + if_cuda_or_rocm([
```

Comment:
why `context/common_context_impl_cuda.cc` and `context/stream_executor_based_impl.cc` are moved out of `if_cuda_or_rocm`

Reply:
These modification follows how `tao_bridge` works in `CMake`. And if not removed out of `if_cuda_or_rocm `, build for cpu device will failed. Also these two files doesn't contain `cuda_or_rocm` device related code, just the wrapping cpp code.

Reply:
After adding related flags, this is no longer needed.